aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -507,6 +502,17 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
   TORCH_CHECK_INDEX(source.dim() == 0 || numIndices == source.size(dim),
           "index_copy_(): Number of indices (", numIndices, ") should be equal to source.size(dim) (", source.size(dim), ")");
 
+  if (self.device().type() == DeviceType::CUDA && globalContext().deterministicAlgorithms()){
```

Comment:
Could you please add the `Note [Enabling Deterministic Operations]` here? https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Context.h#L124

Reply:
thanks, added in the latest version.

Reply:
Cool thanks!